infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -179,6 +186,35 @@ public ServerContext(Map<String, Object> serverConfig) {
         nodeLocator = NodeLocator
                 .parseString(serverConfig.get("--address") + ":" + serverConfig.get("<port>"));
         localEndpoint = nodeLocator.toEndpointUrl();
+        streamNames = loadStreamIdToNameMap((String) serverConfig.get("--stream-names-file"));
+    }
+
+    public static Map<UUID, String> loadStreamIdToNameMap(String streamNameFile) {
+        Map<UUID, String> streamNames = new HashMap<>();
+        if (streamNameFile == null) {
+            return streamNames;
+        }
+        try {
+            BufferedReader csvFile = new BufferedReader(new FileReader(streamNameFile));
```

Comment:
Can we at least use JSON? Since JSON is much easier to work with and we already have JSON parser in our code

Reply:
What capability are we trying to achieve in addition to logging that requires JSON? Once that feature which needs JSON is  added we can certainly log JSON. Right now csv produces the simplest readable stream name mapping file and easy to work on using linux tools.

Reply:
Exactly the same question could be asked regarding csv.
Working with JSON much simpler and easier, than a bunch of meaningless numbers separated by a semicolon and operating with index numbers, also no need for custom code with leaking resources

Reply:
>Not really it's not easy to generate correct JSON by scrapping logs from different sources like client and compactor

how it can work differently for two formats? You do whatever you need to do doesn't matter what the format of your document, and save and read data from JSON is more convenient than from CSV.

>scrapping logs
sounds like a wrong approach. Are you going to do automation stuff by parsing logs by grep commands?

Reply:
can you please share the method to generate correct JSON with streamId -> streamName map without log-scraping while accounting for
1) pseudo-streams like TRANSACTION_STREAM
2) various stream_tags, checkpoint streams
3) ephemeral streams 
4) Internal system tables
5) Log Replicated Streams
from *ALL* the verticals consuming Corfu?

Reply:
the same way how you do that using csv? 
How the file format changes the requirements? Don't you need to do all that magically, if you'd use csv instead of json?

Reply:
I am using a combination of log scrapping and tooling (part of this PR) currently but if you say that is wrong approach, can you please share the correct approach?

Reply:
Again, I'm talking about the file format: `saveToCsv(file)` and `saveToJson(file)` I don't see how `scrappingLogs()` can affect that part

Reply:
Your csv file is just a config, created manually, correct?

Reply:
Yes, what you are suggesting is do another step of post-processing to read CSV and then re-write the tableNames files as JSON.
I will start working on that, but I would like to still understand what is the benefit for this specific use case and not something in the future?